pepr/attack_runner.py

```diff
@@ -0,0 +1,208 @@
+"""
+Attack runner for convenient pentesting with multiple configurations.
+"""
+
+import os
+import logging
+import pickle
+
+import numpy as np
+import tensorflow as tf
+import yaml
+from itertools import groupby
+
+from pepr import report
+from pepr.privacy import mia, gmia
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+def run_attacks(
+    yaml_path, report_save_path, attack_obj_save_path, functions, pdf=False
+):
+    """
+    Run multtiple attacks configured in a YAML file.
+
+    Parameters
+    ----------
+    yaml_path : str
+        Path to attack configuration file in YAML format.
+    report_save_path : str
+        Path where to save the report files.
+    attack_obj_save_path : str
+        Path where to save the serialized attack objects.
+    pdf : bool
+        If true, the attack runner will build the LaTex report and save an PDF to the
+        save path.
+    functions : dict
+        Dictionary containing the functions which return TensorFlow models. The keys
+        should correspond to the strings in the configuration file.
+
+    Returns
+    -------
+    dict
+        Dictionary with attack alias and the corossponding path to the serialized attack
+        object.
+    """
+
+    attack_object_paths = {}
+
+    # Pickle helper
+    def pickle_attack_obj(attack_obj):
+        with open(
+            f"{attack_obj_save_path}/{attack_obj.attack_alias}.pickle", "wb"
+        ) as f:
+            attack_obj.obj_save_path = attack_obj_save_path
+            pickle.dump(attack_obj, f)
+            attack_object_paths[
+                f"{attack_obj.attack_alias}"
+            ] = f"{attack_obj_save_path}/{attack_obj.attack_alias}.pickle"
+
+    # Parse YAML
+    with open(yaml_path) as f_stream:
+        logger.info(f"Parse attack runner configuration file: {yaml_path}")
+        data = yaml.load(f_stream)
```

Comment:
Do you have checked [https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation](https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation)?

Reply:
I wasn't aware of that, thank you for pointing out. The documentation still uses one parameter and I got no deprecation warning during testing, because Colab hat some old 3.x version installed by default :sweat_smile:. 

I added the `SaveLoader` in ff16d50. It should work with 3.14 (Colab default) and 5.4.1 (requirements.txt version).

Reply:
Thank you! :)